src/styles/components/buttons/styles.less

```diff
@@ -24,18 +17,33 @@ components/buttons.less
 
   .button {
 
-    &-enter, &-appear {
+    &-enter,
+    &-appear {
       opacity: 0.01;
       transform: translate(-50%, -100%);
       transition: transform 350ms ease-in, opacity 350ms ease-in;
 
+      &-leave {
+        opacity: 1;
+        transform: translate(-50%, 0);
+      }
+
       &.button {
 
-        &-enter-active, &-appear-active {
+        &-enter-active,
+        &-appear-active {
           opacity: 1;
           transform: translate(-50%, 0);
         }
+
+        &-leave-active {
+          opacity: 0.01;
+          transform: translate(-50%, -100%);
+          transition: transform 350ms ease-in, opacity 350ms ease-in;
+        }
+
       }
+
     }
 
     &.button-short {
```

Comment:
Not quite sure what the intent of `button-short` is and where the the padding values are being derived from.  Seems very arbitrary.